var/spack/repos/builtin/packages/postgresql/package.py

```diff
@@ -14,22 +14,76 @@ class Postgresql(AutotoolsPackage):
+    version('10.4',   sha256='1b60812310bd5756c62d93a9f93de8c28ea63b0df254f428cd1cf1a4d9020048')
     version('10.3', '506498796a314c549388cafb3d5c717a')
     version('10.2', 'e97c3cc72bdf661441f29069299b260a')
-    version('9.3.4', 'd0a41f54c377b2d2fab4a003b0dac762')
+    version('10.1',   sha256='3ccb4e25fe7a7ea6308dea103cac202963e6b746697366d72ec2900449a5e713')
+    version('10.0',   sha256='712f5592e27b81c5b454df96b258c14d94b6b03836831e015c65d6deeae57fd1')
+    version('9.6.12', sha256='2e8c8446ba94767bda8a26cf5a2152bf0ae68a86aaebf894132a763084579d84')
+    version('9.6.11', sha256='38250adc69a1e8613fb926c894cda1d01031391a03648894b9a6e13ff354a530')
     version('9.5.3', '3f0c388566c688c82b01a0edf1e6b7a0')
+    version('9.3.4', 'd0a41f54c377b2d2fab4a003b0dac762')
 
-    depends_on('openssl')
-    depends_on('readline')
-
+    variant('client-only', default=False,
+            description='Build and install client only.')
     variant('threadsafe', default=False, description='Build with thread safe.')
+    variant('lineedit', default='readline',
+            values=('readline', 'libedit', 'none'), multi=False,
+            description='Line editing library')
+    variant('python', default=False, description='Enable Python bindings.')
+    variant('perl', default=False, description='Enable Perl bindings.')
+    variant('tcl', default=False, description='Enable Tcl bindings.')
+    variant('gssapi', default=False,
+            description='Build with GSSAPI functionality.')
 
-    def configure_arg(self):
+    depends_on('readline', when='lineedit=readline')
+    depends_on('libedit', when='lineedit=libedit')
+    depends_on('openssl')
+    extends('tcl', when='+tcl')
+    extends('perl', when='+perl')
+    extends('python', when='+python')
+
+    def configure_args(self):
         config_args = ["--with-openssl"]
+
         if '+threadsafe' in self.spec:
-            config_args.append("--enable-thread-safety")
+            config_args.append('--enable-thread-safety')
         else:
-            config_args.append("--disable-thread-safety")
+            config_args.append('--disable-thread-safety')
+
+        if self.spec.variants['lineedit'].value == 'libedit':
+            config_args.append('--with-libedit-preferred')
+        elif self.spec.variants['lineedit'].value == 'none':
+            config_args.append('--without-readline')
+
+        if '+gssapi' in self.spec:
+            config_args.append('--with-gssapi')
+
+        if '+python' in self.spec:
+            config_args.append('--with-python')
+
+        if '+perl' in self.spec:
+            config_args.append('--with-perl')
+
+        if '+tcl' in self.spec:
+            config_args.append('--with-tcl')
 
         return config_args
+
+    def install(self, spec, prefix):
+        if '+client-only' in self.spec:
+            for subdir in (
+                    'src/bin', 'src/include', 'src/interfaces', 'src/pl'):
```

Comment:
If you want to make this shorter, you can only include the name of the sub-subdir and use `join_path` below since they all exist in `src/`.